Add `Getting started with Ballerina` tutorial (#3433)
ballerina-platform/ballerina-dev-website: learn/getting-started/getting-started-with-ballerina.md

```
layout: getting-started-with-ballerina-left-nav-pages
title: Getting Started with Ballerina

> description: Let’s get you set up a Ballerina development environment and write a simple Ballerina program.

Reviewer: ```suggestion
description: Let’s set up a Ballerina development environment and write a simple Ballerina program.
```

keywords: ballerina, programming language, ballerina packages, getting started
permalink: /learn/getting-started-with-ballerina
active: getting-started-with-ballerina

> intro: Let’s get you set up a Ballerina development environment and write a simple Ballerina program.

Reviewer: ```suggestion
intro: Let’s set up a Ballerina development environment and write a simple Ballerina program.
```

---
```

> Let’s get you set up a Ballerina development environment and write a simple Ballerina program.

Reviewer: ```suggestion
Let’s get you set up a Ballerina development environment and write a simple Ballerina program.
```
We don't need to repeat this here again.

Author: Fixed. Thanks

## Prerequisites

To complete this tutorial, you need:

> - A command terminalA text editor.

Reviewer: ```suggestion
- A command terminal
- A text editor
```

Author: Fixed

> - [VSCode](https://code.visualstudio.com/) with the [Ballerina extension](https://marketplace.visualstudio.com/items?itemName=WSO2.ballerina) installed is our preferred choice, and it has good support for Ballerina.

Reviewer 2: Since we have mentioned about text editor before shouldn't we list these as one is optional. Basically a text editor or VSCode Ballerina extension? We don't need both right?

Author: Fixed. Thanks

- A Ballerina installation. Follow the steps in [Installing Ballerina](https://ballerina.io/learn/installing-ballerina/setting-up-ballerina/).

> - Some experience with at least one programming language. 

Reviewer: ```suggestion
- Some experience with at least one programming language. 
```
Please check the comments added in https://github.com/ballerina-platform/ballerina-dev-website/pull/3369 for the GraphQL guide.

Author: I checked. There is nothing related to this line in that PR.

## Meet `bal`

`bal` is the Ballerina build tool and package manager. Among other things, `bal` helps you to create, build, test, and run your project. 

You can get the latest `bal` tool version with the Ballerina installation. Open your terminal and run the following commands just to make sure everything is ready:

```
$bal
$bal version
```

## Create a new project

> Let’s write a Ballerina program that prints “Hello, World!”. Use the `bal new` command to create a new Ballerina project. 

Reviewer: ```suggestion
Let’s write a Ballerina program that prints “Hello, World!”. Execute the `bal new` command to create a new Ballerina project. 
```

Author: I see nothing wrong with `use` here. IMO, it is better than execute in this context

> ```

Reviewer: ```suggestion
```bash
```
Please update all the occurrences.

$ bal new greeter
```

This command creates a new directory called `greeter`. Let’s move into this directory to see what’s in there. 

```
greeter/
├── Ballerina.toml
└── main.bal
```

- `Ballerina.toml` contains metadata that describes your project. Also, `bal` tool uses `Ballerina.toml` to identify the root of a project.

> - `main.bal` is a source file, and it should contain Ballerina code that prints “Hello, World!” to the console. You can addput any number of source files into the `greeter` directory.

Reviewer: ```suggestion
- `main.bal` is a source file and it should contain Ballerina code that prints “Hello, World!” to the console. You can add any number of source files into the `greeter` directory.
```

## Say “Hello, World!”

You can open the project directory in your text editor. If you are using VSCode, run `code .` inside the `greeter` directory. Then, open `main.bal` to see the generated source.

```ballerina
import ballerina/io;

public function main() {
    io:println("Hello, World!");
}
```

In this code:

> - The first line is an import statement that makes the capabilities in the `[ballerina/io](https://lib.ballerina.io/ballerina/io/latest)` module available to your program. This module contains functions to write to the console, read from the console, and perform read/write operations on files.,

Reviewer: ```suggestion
- The first line is an import statement that makes the capabilities in the `[ballerina/io](https://lib.ballerina.io/ballerina/io/latest)` module available to your program. This module contains functions to write to the console, read from the console, and perform read/write operations on files.
```

Author: Fixed

> - The first line is an import statement that makes the capabilities in the `[ballerina/io](https://lib.ballerina.io/ballerina/io/latest)` module available to your program. This module contains functions to write to the console, read from the console, and perform read/write operations on files.,

Reviewer 2: Links is not properly rendered due to backticks.

> - The `main` function is your program’s entry point, and you can execute it by running the program. 

Reviewer: ```suggestion
- The `main` function is your program’s entry point and you can execute it by running the program. 
```

> - This function contains a statement that prints `Hello, World!` to the console. This statement calls the `println` function in the `io` with `Hello, World!`.

Reviewer: ```suggestion
- This function contains a statement that prints `Hello, World!` in the console. This statement calls the `println` function in the `io` with `Hello, World!`.
```

> If you are interested in learning more about the language, read our [Language Basics](https://ballerina.io/learn/language-basics/) guide. 

Reviewer: ```suggestion
If you are interested in learning more about the language, see [Language Basics](https://ballerina.io/learn/language-basics/). 
```
Changing this for consistency purposes.

> Let’s run this project by running `bal run` in your terminal:

Reviewer: ```suggestion
Let’s run this project by executing `bal run` in your terminal:
```

Author: I prefer running in this context. What's the reason for using execute here?

Reviewer: It is just for consistency purposes we need to use execute and saying `execute` is more formal than saying `run`. However, we can use these 2 interchangably.

```
$ bal run
Compiling source
	example/greeter:0.1.0

Running executable

Hello, World!
```

You can also generate an executable program with `bal build`:

```$ bal buildCompiling source	sameera/greeter:0.1.0
Generating executable	target/bin/greeter.jar

> ```

Reviewer 2: This should be changed as follows. Note: line breaks are not there and here the org name `sameera`  is used while previously `example` 

```
$ bal build
Compiling source
	example/greeter:0.1.0

Generating executable
	target/bin/greeter.jar
```

Author: Fixed

> Then, use `bal run` as follows.

Reviewer: ```suggestion
Then, use `bal run` as follows.
```
Shall we add the instructions as numbered bullet points as they seem sequential?

Reviewer: Will check this in the next review.

```
$ bal run target/bin/greeter.jar
Hello, World!
```

## Write a simple REST API

> Now let’s change our greeter application to a REST API. Ballerina has first-class abstractions for services, resources, etc., and they make network service development easier and more fun. 

Reviewer: ```suggestion
Now, let’s change our `greeter` application to a REST API. Ballerina has first-class abstractions for services, resources, etc., and they make network service development easier and more fun. 
```

You can replace the `main.bal` content with the following code.

> ```

Reviewer: ```suggestion
```ballerina
```
Please fix all occurrences.

Author: Done. Thanks

import ballerina/http;

listener http:Listener httpListener = new (8080);

service / on httpListener {
    resource function get greeting() returns string { 
        return "Hello, World!"; 
    }

    resource function get greeting/[string name]() returns string { 
        return "Hello " + name; 
    }
}
```

Let’s take a moment to digest new constructs in this code:

> - The `[http](https://lib.ballerina.io/ballerina/http/latest)` module provides high-level abstractions to work with the HTTP protocol. 

Reviewer 2: Link is not properly rendering due to back ticks

Author: Fixed

- The listener declaration creates a new HTTP listener with port 8080. Listener is the entity that receives the network input and then routes it to the attached service(s).

> - The service declaration specifies the listener, to which the service gets attached, and a collection of remotely accessible methods. There are two kinds of methods: `resource` and `remote` methods.

Reviewer: ```suggestion
- The service declaration specifies the listener to which the service gets attached and a collection of remotely accessible methods. There are two kinds of methods: `resource` and `remote` methods.
```

> - Services use remote methods to expose services in procedural style: remote methods are named by verbs, and resource methods for data-oriented protocols: resources are named by nouns.

Reviewer: ```suggestion
- Services use remote methods to expose services in procedural style: remote methods are named by verbs, and resource methods for data-oriented protocols: resources are named by nouns.
```
This is not clear.

> - In this example, there are two resource methods: The first one responds to `HTTP` `GET` requests with the path `/greeting`, and the other one responds to `GET` requests with the path `/greeting/{name}`.

Reviewer: ```suggestion
- In this example, there are two resource methods: The first one responds to `HTTP` `GET` requests with the `/greeting` path, and the other one responds to `GET` requests with the `/greeting/{name}` path.
```

> - These resource methods return a `string` value that maps to the content-type `text/plan` in the `HTTP` response.

Reviewer 2: Here `HTTP` is highlighted with backticks, but not in the other places. Can we be consistent ?

Author: Fixed

> If you are interested in learning more about services, read our [Network Interaction](https://ballerina.io/learn/distinctive-language-features/network-interaction/) guide. 

Reviewer: ```suggestion
If you are interested in learning more about services, read our [Network Interaction](https://ballerina.io/learn/distinctive-language-features/network-interaction/) guide. 
```
Instead of saying `if you are interested,` let's directly say as follows.

To learn more about services, see [Network Interaction](https://ballerina.io/learn/distinctive-language-features/network-interaction/).

Author: Fixed

Let’s run this project in your terminal:

```
$ bal run
Compiling source
	example/greeter:0.1.0

Running executable
```

> In another terminal window, try out the following commands:

Reviewer: ```suggestion
In another terminal window, execute the following commands:
```
Let's be consistent with using the word `execute` for commands.

Author: I used `run` instead

```
$ curl localhost:8080/greeting
Hello, World!

$ curl localhost:8080/greeting/Ballerina
Hello Ballerina
```

> ## Do more with Ballerina

Reviewer: ```suggestion
## Do more with Ballerina
```
Can we reword this in a better way? E.g., say `Dive deep` or `Grow with ballerina`?

> In this short tutorial, you set up your development environment and wrote two Ballerina applications.

Reviewer: ```suggestion
In this short tutorial, you set up your development environment and wrote two Ballerina applications.
```
Let's not say `short` tutorial.

Author: Fixed

> To learn more about Ballerina, take a look at the following tutorials:

Reviewer: ```suggestion
To learn more about Ballerina, take a look at the following tutorials:
```
This is more or less a `What's Next` section, which we decided to remove from Learn pages.

Author: I think, we need to have what's next in some situations. This is the getting started guide. Users may want to do more with Ballerina. This point may not be applicable for other guides in the `guides` section
